Updated Note to add RTF and PCE (#689)
mulesoft/docs-connectors: modules/ROOT/pages/sap/sap-connector-terminology.adoc

= SAP Terminology
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

The following are commonly used SAP system acronyms:

ABAP (Advanced Business Application Programming)::
SAP proprietary programming language and part of the NetWeaver platform for building business applications.

ALE (Application Link Enabling)::
Sets up distributed applications and integrated installation of SAP systems. This involves business-driven message exchange using consistent data across loosely linked SAP apps. Apps are integrated through synchronous and asynchronous communication, rather than by use of a central database.

BAPI (Business Application Programming Interface)::
The function module that fulfills design criteria, such as:

> ** Implement a method of a SAP business object

Reviewer: I would change each of these to gerund forms: Implementing, Maintaining, and so on.

** Maintain a static interface through different versions of the SAP system

> ** Enable remotely

Reviewer: This begs for an object: enabling what remotely? Or do we want "Remote enablement"?

Reviewer: ...or even Providing you the ability to...do these things?

Author: Yes, it's remote enablement. I'll figure out a way to reword.

** Run to completion with or without user interaction
** Handle errors

FM (function module)::
Procedures defined in the ABAP language of SAP that allow the encapsulation and reuse of global functions in SAP.

IDoc (intermediate document)::
Standard SAP format for electronic data interchange between SAP systems. Different message types (such as delivery notes or purchase orders) generally correspond to different special formats, known as IDoc types. Multiple message types with related content can, however, be assigned to a single IDoc type.

Java API::

> Handles dynamic metadata lookup and caching. The Java API implements the `JCO.Function` container used for parameters and/or tables for the SAP BAPI function module in Java. Java apps are built on top of the Java API.

Reviewer: GLOBALLY replace "and/or" with the better of the two, per style ref.

JNI (Java Native Interface)::

> JNI is a layer in the SAP Connector library. JCo wraps the C libraries in Java to provide platform-native access into the SAP system. RFC middleware uses the RFC library through the JNI Layer.

Reviewer: Different from other definitions by repeating the term: nec?
Also, should we define JCo?
```suggestion
JNI is a layer in the SAP Connector library. JCo wraps the C libraries in Java to provide platform-native access to the SAP system. RFC middleware uses the RFC library through the JNI layer.
```

RFC (remote function call)::

> Communication with the SAP system is performed over the RFC protocol. RFC means calling BAPI or triggering IDoc processing that runs in another system as a calling program. The RFC interface enables function calls between two SAP systems or between the SAP and external system.

Reviewer: ```suggestion
Communication with the SAP system is performed over the RFC protocol. A remote function call calls BAPI or triggers IDoc processing that runs in another system as a calling program. The RFC interface enables function calls either between two SAP systems or between the SAP and external system.
```

> RFC Layer::

Reviewer: ```suggestion
RFC layer::
```

SAP component that processes remote function calls.

> RFC Library::

Reviewer: ```suggestion
RFC library::
```

Libraries of C language-based functions to access the SAP system. The RFC library is addressed by JNI.

SAP Java IDoc class library::
Provides structured, high-level interpretation and navigation of SAP IDocs in Java that consists of the following add-on packages:

> ** SAP Java base IDoc class library: a middleware-independent library that provides a set of general base classes and interfaces for middleware dependent Java IDoc class library implementations.

Reviewer: Java-based?
```suggestion
** SAP Java base IDoc class library: A middleware-independent library that provides a set of general base classes and interfaces for middleware-dependent Java IDoc class library implementations.
```

** SAP Java connector IDoc class library: A middleware-independent library for creating, sending, and receiving IDocs.

SAP NetWeaver::

> One of the main technologies and application platforms used by SAP solutions. The main component of NetWeaver is the SAP Web Application Server (WebAS), which provides the runtime environment for SAP applications like ERP, CRM, SCM, PLM, SRM, and BI. Other components include the enterprise portal, exchange infrastructure, master data management, and mobile infrastructure. The SAP NetWeaver is an umbrella term for these technical components. SAP NetWeaver runs on both the Java and ABAP stacks.

Reviewer: ```suggestion
One of the main technologies and application platforms used by SAP solutions. The main component of NetWeaver is SAP Web Application Server (WebAS), which provides the runtime environment for SAP applications like ERP, CRM, SCM, PLM, SRM, and BI. Other components include the enterprise portal, exchange infrastructure, master data management, and mobile infrastructure. SAP NetWeaver is an umbrella term for these technical components. SAP NetWeaver runs on both the Java and ABAP stacks.
```

> SNC (Secure Network Communications)::

Reviewer: I see both "Secure" and "Secured" when I search online. What do you think?

Author: Secure.

> SNC integrates with the SAP NetWeaver single sign-on, or an external security product that works with SAP systems. SNC provides security functions that are not directly available with SAP systems.

Reviewer: ditto re. renaming term:
```suggestion
Integrates with SAP NetWeaver single sign-on or an external security product that works with SAP systems. SNC provides security functions that are not directly available with SAP systems.
```
